heron/stmgr/src/cpp/manager/tmaster-client.h

```diff
@@ -86,7 +84,8 @@ class TMasterClient : public Client {
   sp_string stmgr_host_;
   sp_int32 stmgr_port_;
   sp_int32 shell_port_;
-  std::vector<proto::system::Instance*> instances_;
+  proto::tmaster::StMgrRegisterRequest* register_request_;
```

Comment:
why not make it a an actual instance of register request rather than a pointer? That way the cleaning logic is greatly simplified

Reply:
Good suggestion. The original reason I want to use pointer here is I can do safety check before sending out the request. Now I updated the code: 1. use an instance instead of pointer; 2. introduce a boolean value indicating if the request is initialized properly  for safety check